cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java

```diff
@@ -150,23 +169,125 @@ private Action parseAction(String action) {
     }
   }
 
-  private void migrateMetricsData(Injector injector) throws Exception {
+
+  private void migrateMetricsData(Injector injector, boolean keepOldTables) throws Exception {
+    CConfiguration cConf = injector.getInstance(CConfiguration.class);
+    Configuration hConf = injector.getInstance(Configuration.class);
+
     System.out.println("Starting metrics data migration");
     // find version to migrate
     MetricHBaseTableUtil metricHBaseTableUtil = new MetricHBaseTableUtil(injector.getInstance(HBaseTableUtil.class));
     Version version = findMetricsTableVersion(injector.getInstance(CConfiguration.class),
                                               injector.getInstance(Configuration.class), metricHBaseTableUtil);
+
     if (version != null) {
       System.out.println("Migrating Metrics Data from " + version.name());
       DatasetFramework framework = createRegisteredDatasetFramework(injector);
+      // perform sanity check - truncate 2.8 tables data
+      truncateV2Tables(cConf, hConf);
```

Comment:
One thing (feel free to create separate JIRA): we want to fail by default if destination tables are not clean, unless user specifies argument --cleanup-destination or smth. I.e. if someone runs migration by mistake - it'll tell that destination tables exist and not empty, are you sure you want to delete them and migrate old data over? use --foo option to skip this check in that case. Smth like that